### ISSUE TITLE
What is the intuition of flipping training labels continually?

### ISSUE THREAD

**Issue by gordon-lim** [VTCSML/Adversarial-Label-Learning#1]
Dear Dr. Arachie,

Thank you for your work here. I thinks its an interesting idea. Intuitively, I'm struggling a little bit to wrap my head around the intuition of the adversary's role. Am I right that during the same training procedure, an image (for example) could be presented to the model with different labels across epochs? Say for example in CIFAR-10, a multi-class image classification task, an image has the weak labels _dog_ and _cat_. _Cat_ maximises the classifier's error so we pick that label. Later on, _dog_ maximises the classifier's error so we pick that label. Am I right on this? Intuitively, this does not like conducive learning. At the same time I'm also reminded of [MixUp](https://towardsdatascience.com/enhancing-neural-networks-with-mixup-in-pytorch-5129d261bc4a) augmentation with soft labels. I know its been 5 years but I would appreciate deeply if you could enlighten me. Also, I'm curious what is your opinion of how your paradigm fits into today's landscape with other methods in the Learning with Noisy Labels (LNL) / weak-supervision space. Is there any particular reason why you have not pursued this research direction further? Perhaps your insights can inspire my own research.

Yours Sincerely,
Gordon
